File: company/facebook/338.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
	vector<int> countBits(int num) {
		vector<int> res(num+1,0);
		for(int n=1;n<=num;++n)
			res[n] = res[n >> 1] + (1 & n);
		return res;

	}
};

File: Python/338.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        res = [0]*(num+1)
        for i in range(1, num+1):
            res[i] = res[i&(i-1)]+1
        return res
